### DISCUSSION TITLE
TO - STATUS

### DISCUSSION THREAD

**Discussion by SantiagoOsorio3** [IIC2133-PUC/2024-2#15] [Tarea 0] [answered]
Hola, tengo una pregunta en como funciona especificamente fscanf, esto ya que estoy en la parte de crear STATUS y me esta pasando que se imprime todo perfectamente pero no se lee lo que viene después en el input (esto nose si es que porque al poner (string_equals(command, "STATUS")) se esta esperando algo mas ?? 

Tambien trate de debbugear esto, en donde agregue abajo un fscanf en donde efectivamente se me guarda la linea de abajo. 

En resumen, mi duda va en si es que al poner solo (string_equals(command, "STATUS")) sin fscanf dentro, el codigo espera una variable infinitamente ? o el error debe ser otra cosa ? (cabe recalcar que mi STATUS funciona perfectamente, asi la ejecucion no se detiene dentro de esta)

**Comment by tomastrivino** [comment, marked as answer]
Hola Santiago,
acabo de revisar tu código.
Lo que hace este for loop es iterar para la cantidad de eventos y leer una línea usando `fscanf`. Luego, dependiendo del comando que se entregue defines cómo seguir el flujo (si es STATUS, ENTER, etc.).
```c
for (int i = 0; i < E; i++)

  { 
    fscanf(input_file, "%s", command);
   // Esta línea es la que recibe el comando, por lo que hacemos string_equals(command, "EVENTO").
   // De ahí que no tenga el fscanf dentro del string_equals. En resumen, el error es otra cosa :c
     .... 
   }

```
Consejos
---
- Puedes modularizar las partes: es decir, no es necesario hacer todo en tu `main.c`. Quizás podrías encapsular lo que haces para cada comando en funciones e incluirlas en otros módulos.
- Podrías usar variables más descriptivas: usar `i`, `j` y `k` puede llevar a confusiones, en especial cuando hay `loops` anidados.

¡éxitooo!
![image](https://github.com/user-attachments/assets/1cd1adac-92e2-4dad-88a6-4bac09446a5a)

**Reply by SantiagoOsorio3** [reply]
Muchas gracias ! Ahi pude arreglarlo, era un for que tenia i = 0 en vez de int i = 0, en donde supongo que me habia aparecido el error en algun make, pero no me di cuenta y por eso tiraba error.
